blocks/hooks/test/autocompleters.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import { mount, shallow } from 'enzyme';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter, removeFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { filterAutocomplete } from '../autocompleters';
+import defaultCompleters from '../../autocompleters';
+
+function TestComponent() {
+	// Use a naturally focusable element because we will test with focus.
+	return <input />;
+}
+const FilteredComponent = filterAutocomplete( TestComponent );
+
+describe( 'autocompleters', () => {
+	afterEach( () => {
+		removeFilter( 'components.Autocomplete.completers', 'test/autocompleters-hook' );
+	} );
+
+	it( 'filters supplied completers when next focused', () => {
+		const completersFilter = jest.fn();
+		addFilter(
+			'components.Autocomplete.completers',
+			'test/autocompleters-hook',
+			completersFilter
+		);
+
+		const expectedCompleters = [ {}, {}, {} ];
+		const wrapper = mount( <FilteredComponent completers={ expectedCompleters } /> );
+
+		try {
+			expect( completersFilter ).not.toHaveBeenCalled();
+			wrapper.find( 'input' ).simulate( 'focus' );
+			expect( completersFilter ).toHaveBeenCalledWith( expectedCompleters );
+		} finally {
+			wrapper.unmount();
```

Comment:
It'd be less code to put `wrapper.unmount()` inside `afterEach` call instead of using try / finally in all tests.

Reply:
I was actually doing it that way initially but switched away because I thought I would be able to use `shallow` for most of the tests. I was wrong. :)

I started to prefer the explicit try/finally because it is easy to accidentally declare a local `const wrapper` that shadows the outer variable, but I just noticed we have a `no-shadow` lint rule that covers this case.

I'll switch it back.